### DISCUSSION TITLE
How could I calculate the volume size?

### DISCUSSION THREAD

**Discussion by DiMiTriFrog** [ejtraderLabs/ejtraderCT#26] [Q&A]
I'm trying to calculate the volume size by a function i found in ctrade forum:

```
def get_lot_size(pair, balance, pip_loss, price_open, risk=0.5,):
    pip_scale = float(get_pip_size(pair))
    amount_to_risk = (float(balance) * (float(risk)/100))
    trade_volume = amount_to_risk / pip_loss * pip_scale
    truncation_factor = 100000 * pip_scale * 100

    trade_volume =  int(trade_volume / truncation_factor) * truncation_factor
    return truncation_factor

```

but now, i'm have this output: 
**ERROR:root:Order volume 100000000.00 is bigger than maxVolume=5000000.00**

Has someone help. me in this step?

Thanks :)

**Comment by traderpedroso** [comment]
> 

Sorry, but this is a question somewhat for Stack Overflow, but it's okay. The calculation must be based on the asset being traded. For example, assets with 5 digits after the decimal point, like EURUSD have a pip size of 0.0001. Whereas assets like XAUUSD have 2 digits after the decimal point, and the pip size is 0.01.


``` python
def get_pip_size(pair):
    if 'JPY' in pair.upper():
        return 0.01
    else:
        return 0.0001

```

**Reply by DiMiTriFrog** [reply]
But I want to mean te volume size of the buy function. The Lots.

**Comment by traderpedroso** [comment]
> But I want to mean te volume size of the buy function. The Lots.

I'm sorry, my friend, but the API is already handling this. If you just want to add a specific size, for example, 0.10 on EURUSD, you just need to consult the documentation. This is the minimum effort required on your part. The code in the API is essentially doing the following operation: size = int(float(v_lots) * 100000).

**Reply by DiMiTriFrog** [reply]
Thanks!
And how could I calculate the v_lots?
